tools/generators/legacy/src/DTO/XcodeScheme.swift

```diff
@@ -36,15 +36,22 @@ extension XcodeScheme {
     }
 
     /// Create a new scheme applying any default actions based upon the current scheme.
-    var withDefaults: XcodeScheme {
-        get throws {
-            var buildTargets = [BazelLabel: XcodeScheme.BuildTarget]()
-
-            func enableBuildForValue(
-                _ label: BazelLabel,
-                _ keyPath: WritableKeyPath<XcodeScheme.BuildFor, XcodeScheme.BuildFor.Value>,
-                _ mode: EnableBuildForValueMode = .merge
-            ) throws {
+    func withDefaults(
+        aliasLabels: [BazelLabel: [BazelLabel]] = [:]
+    ) throws -> XcodeScheme {
+        var buildTargets = [BazelLabel: XcodeScheme.BuildTarget]()
+
+        func enableBuildForValue(
+            _ targetLabel: BazelLabel,
+            _ keyPath: WritableKeyPath<XcodeScheme.BuildFor, XcodeScheme.BuildFor.Value>,
+            _ mode: EnableBuildForValueMode = .merge
+        ) throws {
+            var labelsToProcess: [BazelLabel] = [targetLabel]
+            if let actualLabels = aliasLabels[targetLabel], actualLabels.count > 0 {
+                labelsToProcess = actualLabels
+            }
```

Comment:
Can `actualLabels.count` ever be 0?

Reply:
(I guess with my suggestion we would want this check anyway.)

Reply:
If the way we collect it works as expected no. This is just me being a bit paranoid maybe. I guess we could improve readability here by removing it since if this ever gets here empty it should be a visible failure in the generator.